### ISSUE TITLE
Specific World creates memory overflow on only Headless Server

### ISSUE THREAD

**Issue by Readun** [Yellow-Dog-Man/Resonite-Issues#2213]
### Describe the bug?

I have a specific old world that seem to rapidly overflow the memory until it runs out of it, leading to a crash.
The Headless crashes in about~ 1 Minute and is still joinable in that timeframe (But it has low fps)

Interestingly it also has a high CPU Usage, similiar to [https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2079](url) & [,](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1410) but that one doesnt have the memory overflow issue and doesnt crash, if the cpu is strong enough.

It also sometimes show similiar exeptions like [https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1198](url) but im not fully sure if it is the same. (Different results in exeptions, see Screenshoots)

### To Reproduce

I only need to load my "Dragoncave" world, that I imported from the old Platform. It worked on my Headless before.
Would provide the record ID more discreetly.

### Expected behavior

Run the world without massive CPU & Memory usage.

### Screenshots

![Screenshot 2024-06-02 162107](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/5b1ed0a8-ef69-4c1d-b34d-f862eaf45fb8)
![Screenshot 2024-06-02 162053](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/f6130519-4980-44bc-a752-7c976a6463cc)
![Screenshot 2024-06-02 162126](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/3885fb9e-cc87-4347-b23f-a8094938f120)
![Screenshot 2024-06-02 162518](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/47f8c065-bd3f-4ecc-8d7b-f8e3f78e20d3)

### Resonite Version Number

2024.5.31.202 (And as far as I know, since release)

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Headless

### Log Files

[READUNSERVER - 2024.5.31.202 - 2024-06-02 16_26_05.log](https://github.com/user-attachments/files/15525673/READUNSERVER.-.2024.5.31.202.-.2024-06-02.16_26_05.log)
[READUNSERVER - 2024.5.31.202 - 2024-06-02 16_19_28.log](https://github.com/user-attachments/files/15525674/READUNSERVER.-.2024.5.31.202.-.2024-06-02.16_19_28.log)

### Additional Context

It works normally when opening the world with the normal client (Screenmode & SteamVR)

### Reporters

readun

**Comment by Frooxius** [comment]
Are you able to isolate this? Delete stuff from the world until it stops happening, then delete other stuff as long as it keeps happening.

**Comment by Readun** [comment]
Can try when I have time, i already deleted all old video players to replace them with new too, but didn't helped.
(Only one remains)

**Comment by Readun** [comment]
Im currently trying to debug the world. But I found out this is a more major bug...
The cpu load and occupied ram will also -stay- after you leave the world... I joined the world 3 times now and my desktop cpu is pegged at 100% while in my home world.
first time -> 60%
second time -> 80%
third time -> 100%

the ram usage stayed too and increased with each round.

![opening the world 3 times](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/8c49422a-4d86-4d4c-899e-40b00b210ecd)

and it stayed after leaving them too.
![persisting cpu load](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/101413aa-08dd-4c7d-a03c-78aefc740dc8)

third time:

![3 times](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/fedd5905-81d1-4b6c-8f16-0b3de513e337)

forth time:
![4 times](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/eb0ef622-1e16-4904-abc3-1aeded256eb5)
![more ram](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/a9905215-de7f-4e1c-b36f-0058cdc23045)

I did deleted some reparent flux and some videoplayers, the server this time crashed, without running out of memory. But it definatly still had a memory leak!:
![server ram slow](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/cc33a593-a401-49db-b673-e24f7c951bf2)
Server Log:

[READUNSERVER - 2024.5.31.202 - 2024-06-02 23_13_19.log](https://github.com/user-attachments/files/15526901/READUNSERVER.-.2024.5.31.202.-.2024-06-02.23_13_19.log)


I will continue to strip down the world for now, to get a clue what it could be. But it miight have something to do with videos again?

But this run makes me question, how can the cpu and ram load stay after leaving a world?

**Comment by Readun** [comment]
Addition... When being in my local home:

![local home](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/ab39a8f4-7d8d-44a9-ba38-dbfe37842887)

**Comment by Readun** [comment]
@Frooxius  After testing it a couple of times, I can now confirm -> Convex Hulls
Convex hulls the root of that issue, my world had a handcrafted cave and stonehege out of convex hulls (So a ton of them) and it caused the memory leaks & Headless crash in this case.
After Deleting them all, the Headless now runs on 0-1%
![grafik](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/6e9385be-9d82-421b-bbb7-3205a6e408d6)


The Log of the "Fixed world" But tbf, I cant be sure if I left some convex hulls somewhere...

[READUNSERVER - 2024.5.31.202 - 2024-06-03 00_18_00.log](https://github.com/user-attachments/files/15527106/READUNSERVER.-.2024.5.31.202.-.2024-06-03.00_18_00.log)


So #1198 seems to not actually be fixed & it will stay as long as Resonite runs, not going away by just leaving the world.

**Comment by shiftyscales** [comment]
Please include a link to a copy of the world with everything stripped away except the problematic assets as indicated above by Frooxius, @Readun.

**Comment by Readun** [comment]
Got the Convex hulls now into a gridspace and they are doing the exact same behavior:
World Record: resrec:///U-Readun/R-ca8dcfd8-3ae1-4a8c-8d9a-4cbbf2fed102

![grafik](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76232047/e3021aea-9f0d-4d18-87f9-adb894263735)

**Comment by shiftyscales** [comment]
Thanks- this issue seems to have everything it should need to be looked further into at some point, @Frooxius.

**Comment by Readun** [comment]
Something Additional: 
I noticed, even when deleting the massive amount of Convex hulls and -just- having the 7 "Stones" at spawn, they still cause that huge CPU load.
When I was walking around, some of them are not having a collision, while Character collider is toggled.
(So the System seems to be in a loop of trying to calculate the colider?)

![Image](https://github.com/user-attachments/assets/59aa4b27-004c-41cd-b868-930dc1d562ba)

**Comment by Frooxius** [comment]
Yeah it sounds like it's likely stuck in an infinite loop that never finishes. 

We'll have to make a reproduction item for BEPUv2 to fix it up and then backport the changes.

Although once we upgrade to .NET 9, we'll sync up with latest master, which might already have the fixes? I don't know how urgent this is, if it's something you can work around for the time being.
